# HG changeset patch
# User Rik <rik@octave.org>
# Date 1372638606 25200
#      Sun Jun 30 17:30:06 2013 -0700
# Node ID cc3743a91652c2a8e321abde8f3a31e8bfbce375
# Parent  011119bf48681afcaf53bba9015a1b67830b701c
Rename read_readline_init_file to readline_read_init_file.


* libinterp/interpfcn/input.cc(Fread_readline_init_file): Rename function.

* doc/interpreter/basics.txi: Change function name in manual.

* libgui/src/resource-manager.cc: Rename function in function list.

* scripts/startup/main-rcfile: Call readline_read_init_file on startup.

* scripts/deprecated/read_readline_init_file.m: Deprecated function which
calls readline_read_init_file.
* scripts/deprecated/module.mk: Add m-file to build system.

diff --git a/doc/interpreter/basics.txi b/doc/interpreter/basics.txi
--- a/doc/interpreter/basics.txi
+++ b/doc/interpreter/basics.txi
@@ -744,17 +744,17 @@ Octave uses the GNU Readline library for
 history features.  Readline is very flexible and can be modified through
 a configuration file of commands (See the GNU Readline library for the
 exact command syntax).  The default configuration file is normally
 @file{~/.inputrc}.
 
 Octave provides two commands for initializing Readline and thereby changing
 the command line behavior.
 
-@DOCSTRING(read_readline_init_file)
+@DOCSTRING(readline_read_init_file)
 
 @DOCSTRING(re_read_readline_init_file)
 
 @node Customizing the Prompt
 @subsection Customizing the Prompt
 @cindex prompt customization
 @cindex customizing the prompt
 
diff --git a/libgui/src/resource-manager.cc b/libgui/src/resource-manager.cc
--- a/libgui/src/resource-manager.cc
+++ b/libgui/src/resource-manager.cc
@@ -1384,18 +1384,18 @@ resource_manager::octave_keywords (void)
       "range "
       "rank "
       "ranks "
       "rat "
       "rats "
       "rcond "
       "rdivide "
       "re_read_readline_init_file "
-      "read_readline_init_file "
       "readdir "
+      "readline_read_init_file "
       "readlink "
       "real "
       "reallog "
       "realmax "
       "realmin "
       "realpow "
       "realsqrt "
       "record "
diff --git a/libinterp/interpfcn/input.cc b/libinterp/interpfcn/input.cc
--- a/libinterp/interpfcn/input.cc
+++ b/libinterp/interpfcn/input.cc
@@ -1060,19 +1060,19 @@ a feature, not a bug.\n\
         }
     }
   else
     print_usage ();
 
   return retval;
 }
 
-DEFUN (read_readline_init_file, args, ,
+DEFUN (readline_read_init_file, args, ,
   "-*- texinfo -*-\n\
-@deftypefn {Built-in Function} {} read_readline_init_file (@var{file})\n\
+@deftypefn {Built-in Function} {} readline_read_init_file (@var{file})\n\
 Read the readline library initialization file @var{file}.  If\n\
 @var{file} is omitted, read the default initialization file (normally\n\
 @file{~/.inputrc}).\n\
 \n\
 @xref{Readline Init File, , , readline, GNU Readline Library},\n\
 for details.\n\
 @end deftypefn")
 {
@@ -1096,16 +1096,17 @@ for details.\n\
 }
 
 DEFUN (re_read_readline_init_file, args, ,
   "-*- texinfo -*-\n\
 @deftypefn {Built-in Function} {} re_read_readline_init_file ()\n\
 Re-read the last readline library initialization file that was read.\n\
 @xref{Readline Init File, , , readline, GNU Readline Library},\n\
 for details.\n\
+@seealso{readline_read_init_file}\n\
 @end deftypefn")
 {
   octave_value_list retval;
 
   if (args.length () == 0)
     command_editor::re_read_init_file ();
   else
     print_usage ();
diff --git a/scripts/deprecated/module.mk b/scripts/deprecated/module.mk
--- a/scripts/deprecated/module.mk
+++ b/scripts/deprecated/module.mk
@@ -14,16 +14,17 @@ deprecated_FCN_FILES = \
   deprecated/java_get.m \
   deprecated/java_invoke.m \
   deprecated/java_new.m \
   deprecated/java_unsigned_conversion.m \
   deprecated/java_set.m \
   deprecated/javafields.m \
   deprecated/javamethods.m \
   deprecated/polyderiv.m \
+  deprecated/read_readline_init_file.m \
   deprecated/saving_history.m \
   deprecated/shell_cmd.m \
   deprecated/studentize.m \
   deprecated/sylvester_matrix.m
 
 FCN_FILES += $(deprecated_FCN_FILES)
 
 PKG_ADD_FILES += deprecated/PKG_ADD
diff --git a/scripts/deprecated/read_readline_init_file.m b/scripts/deprecated/read_readline_init_file.m
new file mode 100644
--- /dev/null
+++ b/scripts/deprecated/read_readline_init_file.m
@@ -0,0 +1,40 @@
+## Copyright (C) 2013 Rik Wehbring
+##
+## This file is part of Octave.
+##
+## Octave is free software; you can redistribute it and/or modify it
+## under the terms of the GNU General Public License as published by
+## the Free Software Foundation; either version 3 of the License, or (at
+## your option) any later version.
+##
+## Octave is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Octave; see the file COPYING.  If not, see
+## <http://www.gnu.org/licenses/>.
+
+## -*- texinfo -*-
+## @deftypefn {Built-in Function} {} read_readline_init_file (@var{file})
+## This function has been deprecated.  Use
+## @code{@file{readline_read_init_file}} instead.
+## @seealso{readline_read_init_file}
+## @end deftypefn
+
+## Deprecated in 3.8
+
+function read_readline_init_file (varargin)
+
+  persistent warned = false;
+  if (! warned)
+    warned = true;
+    warning ("Octave:deprecated-function",
+             "read_readline_init_file is obsolete and will be removed from a future version of Octave, please use readline_read_init_file instead");
+  endif
+
+  readline_read_init_file (varargin{:});
+
+endfunction
+
diff --git a/scripts/startup/main-rcfile b/scripts/startup/main-rcfile
--- a/scripts/startup/main-rcfile
+++ b/scripts/startup/main-rcfile
@@ -1,17 +1,17 @@
 ## System-wide startup file for Octave.
 ##
 ## This file should contain any commands that should be executed each
 ## time Octave starts for every user at this site.
 
 ## Configure readline using the file inputrc in the Octave startup
 ## directory.
 
-read_readline_init_file (sprintf ("%s%s%s",
+readline_read_init_file (sprintf ("%s%s%s",
 				  octave_config_info ("startupfiledir"),
 				  filesep, "inputrc"));
 
 if (strcmp (PAGER (), "less") && isempty (getenv ("LESS")))
   PAGER_FLAGS ('-e -X -P"-- less ?pB(%pB\\%):--. (f)orward, (b)ack, (q)uit$"');
 endif
 
 ## This appears here instead of in the pkg/PKG_ADD file so that --norc
